embassy-stm32/src/flash/mod.rs

```diff
@@ -17,74 +19,175 @@ pub struct Flash<'d> {
     _inner: PeripheralRef<'d, FLASH>,
 }
 
+static REGION_LOCK: Mutex<CriticalSectionRawMutex, ()> = Mutex::new(());
+
 impl<'d> Flash<'d> {
     pub fn new(p: impl Peripheral<P = FLASH> + 'd) -> Self {
         into_ref!(p);
         Self { _inner: p }
     }
 
+    pub fn into_regions(self) -> FlashRegions {
```

Comment:
The lifetime here needs to carry over into `FlashRegions` and then into the individual regions.

Otherwise you can create a `Flash` with borrowed (`&mut`) peripherals, then convert it to `FlashRegions`, which releases the borrow because the lifetime is no longer active, then create *another* `Flash` without dropping the `FlashRegions` first.

Reply:
Ahh, I did not realize that p could be a reference. This is now fixed.